### DISCUSSION TITLE
Text encoding by default

### DISCUSSION THREAD

**Discussion by Buzut** [pocketbase/pocketbase#5400] [Q&A] [answered]
Hey there,
I confess I haven't manually checked the database, I know it's in the headers. That being said, nowhere in the docs is the default text encoding mentioned. I assume it's UTF-8, but it would be a plus to mention it somewhere in the docs as text encoding divergence would make searching and sorting broken.

In Asia, UTF-16 is sometimes preferred to UTF-8, so maybe even offer the encoding options on first install?

**Comment by ganigeorgiev** [comment, marked as answer]
Go by default assume that all string literals hold valid UTF8 sequences, which Go call "runes" (_unless there is byte-level escapes, but that's another rabbit hole I don't want to get into_).

SQLite also uses by default UTF-8 encoding 

There are no plans to change the encoding as this could lead to a lot of complexity and will require special handling practically everywhere where we have string operations (for the SQLite Go driver this may even require explicitly specifying what function to use when opening the connection, aka. `sqlite3_open_v2` vs `sqlite3_open16`, which will further complicate the task).

**Reply by Buzut** [reply]
Thanks for getting back to me promptly. I am absolutely cool with that as UTF-8 is what I use 99,99% of the time. I was just curious about it and think a quick mention in the docs wouldn't hurt.
Might propose a PR if you'd like me to, just to clear things out.

**Reply by ganigeorgiev** [reply]
No, it would be confusing and I don't see why it should matter for the end user. UTF8 is a variable-length encoding and a character could be anywhere between 1-4 bytes.

Yes, it could eventually take more disk space for some characters and could be slightly slower to decode but again, for the end user this shouldn't matter because it is happening transparently.

If you have stumbled on a specific limitation feel free to elaborate but as of now I don't understand what is the issue here.

**Reply by Buzut** [reply]
Sorry for the late reply. 

Basically, UTF8 is expected, so it doesn't hurt to mention it. Not mentioning it anywhere could imply that encoding is either auto-detected or doesn't have any importance, which isn't the case.

If data is sent over with a different encoding, it'll be stored as is in the database and will show up broken in the back office. This could come from a webpage encoding other than UTF-8 or data from source code with a different encoding.

ISO8859-1 (a.k.a Latin-1) and its derivatives are still widespread, and it'll break the encoding with anything other than ASCII. Quick php exemple sending data over to the REST API? Save the file in Latin-1 and admire the result.

```php
<?php

$bearer = '';

// Set cURL options
$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, 'http://127.0.0.1:8090/api/collections/test/records');
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, [
    'text' => "C'est comme ça qu'on fait nous."
]);

// Add the Bearer Token to the Authorization header
curl_setopt($ch, CURLOPT_HTTPHEADER, [
    "Authorization: Bearer $bearer"
]);

// Execute the cURL session
$response = curl_exec($ch);

// Check for errors
if (curl_errno($ch)) {
    echo 'Error: ' . curl_error($ch) . "\n";
} else {
    echo "Server Response: " . $response . "\n";
}

// Close the cURL session
curl_close($ch);
```

![Screenshot 2024-09-13 at 22 39 52](https://github.com/user-attachments/assets/2dd9bcc0-9379-4159-845b-f91a8e62d6c8)
